packages/components/src/tools-panel/stories/tools-panel-with-item-group-slot.js

```diff
@@ -0,0 +1,246 @@
+
+	const resetAll = ( resetFilters = [] ) => {
+		let newAttributes = {};
+
+		resetFilters.forEach( ( resetFilter ) => {
+			newAttributes = {
+				...newAttributes,
+				...resetFilter( newAttributes ),
+			};
+		} );
+
+		setAttributes( newAttributes );
+	};
+
+	const updateAttribute = ( name, value ) => {
+		setAttributes( {
+			...attributes,
+			[ name ]: value,
+		} );
+	};
+
+	const ToolsPanelColorDropdown = ( { attribute, label, value } ) => {
+		return (
+			<ToolsPanelItem
+				className={ itemClassName }
+				hasValue={ () => !! value }
+				label={ label }
+				onDeselect={ () => updateAttribute( attribute, undefined ) }
+				resetAllFilter={ () => ( { [ attribute ]: undefined } ) }
+				panelId={ panelId }
+				as={ Item }
+			>
+				<Dropdown
+					renderToggle={ ( { onToggle } ) => (
+						<Button onClick={ onToggle }>
+							<HStack justify="flex-start">
+								<ColorIndicator colorValue={ value } />
+								<FlexItem>{ label }</FlexItem>
+							</HStack>
+						</Button>
+					) }
+					renderContent={ () => (
+						<ColorPalette
+							value={ value }
+							colors={ colors }
+							onChange={ ( newColor ) =>
+								updateAttribute( attribute, newColor )
+							}
+						/>
+					) }
+				/>
+			</ToolsPanelItem>
+		);
+	};
+
+	// ToolsPanelItems are rendered via two different fills to simulate
+	// injection from multiple locations.
+	return (
+		<SlotFillProvider>
+			<PanelWrapperView>
+				<Panel>
+					<ToolsPanelItems.Slot
+						as={ ItemGroup }
+						className={ slotWrapperClassName }
+						resetAll={ resetAll }
+					/>
+				</Panel>
+			</PanelWrapperView>
+			<ToolsPanelItems>
+				<ToolsPanelColorDropdown
+					attribute="text"
+					label="Text"
+					value={ text }
+				/>
+				<ToolsPanelColorDropdown
+					attribute="background"
+					label="Background"
+					value={ background }
+				/>
+			</ToolsPanelItems>
+			<ToolsPanelItems>
+				<ToolsPanelColorDropdown
+					attribute="link"
+					label="Link"
+					value={ link }
+				/>
+			</ToolsPanelItems>
+		</SlotFillProvider>
+	);
+};
+
+const PanelWrapperView = styled.div`
+	max-width: 280px;
+	font-size: 13px;
+
+	.components-dropdown-menu__menu {
+		max-width: 220px;
+	}
+`;
+
+const SlotWrapper = css`
+	&&& {
+		row-gap: 0;
+	}
+
+	> div {
+		grid-column: span 2;
+	}
+`;
+
+const ToolsPanelItemClass = css`
+	padding: 0;
+	border-left: 1px solid rgba( 0, 0, 0, 0.1 );
+	border-right: 1px solid rgba( 0, 0, 0, 0.1 );
+	border-bottom: 1px solid rgba( 0, 0, 0, 0.1 );
+
+	&&.first {
+		border-top-left-radius: 10px;
+		border-top-right-radius: 10px;
+		border-top: 1px solid rgba( 0, 0, 0, 0.1 );
+	}
+
+	&.last {
+		border-bottom-left-radius: 10px;
+		border-bottom-right-radius: 10px;
+	}
+
+	&& > div,
+	&& > div > button {
+		width: 100%;
+		border-radius: inherit;
+	}
+
+	/* .components-dropdown class overrides ToolsPanelItemPlaceholder styles */
+	&[class*='ToolsPanelItemPlaceholder'] {
+		display: none;
+	}
```

Comment:
Let's not use this class selector — these classnames are computed at runtime and in production are hashed.

Also, it looks like (with the latest changes) this override is not needed anymore?

Reply:
Yes, it is no longer needed. Previously, it was only an issue in the storybook not the block editor etc.